nixos: fix avahi connectivity for shairport-sync module

anderslundstedt opened the pull request:
The shairport-sync service currently fails to start with the error

`shairport avahi_entry_group_new failed`

This problem seems to have been introduced by cdd7310a503481e3c40266be45b6b8256d95ecbd. After some trial and error I concluded that the attached commit is a minimal fix.

bjornfor commented:
IMHO, it'd be nice to have your PR cover message in the commit message, because it explains the "why" and adds traceability ("it fixes [this] since [that] broke...).

(Btw, did you know that when you create a PR containing a single commit, the commit message will be copied to the PR cover message? So if you start with the commit, you don't have to write that PR cover...)

anderslundstedt committed:
nixos: fix avahi connectivity for shairport-sync module

The shairport-sync service currently fails to start with the error

shairport avahi_entry_group_new failed

This problem seems to have been introduced by

cdd7310a503481e3c40266be45b6b8256d95ecbd

After some trial and error I concluded that the attached commit is a minimal
fix.

anderslundstedt commented:
@bjornfor fixed.

bjornfor commented:
@anderslundstedt: Thanks! Pushed to master (5f3c4bd11ebdd88d56dbf81ebb997832c7b82f3b). I'll backport to release-16.03 as soon as the current eval-blocker is fixed (https://github.com/NixOS/nixpkgs/commit/094c6ceff0a9a0ac2bdf2c4e16a14fc58ac7ec3f#commitcomment-18174818).

bjornfor commented:
And now also in release-16.03 (6320299e1670d007ad126d3024b2705d7a52a816).
